### PR TITLE
Add example with parallel assignment.

### PR DESCRIPTION
**Pull request by majjoha**
Using a combination of parallel assignment and the splat operator, we can easily split the array into the head and tail. I think this example is worth including along with the other examples. It is not better than `#drop`, but it is simply just yet another way of doing it.

### CODE REVIEW

**Comment by croaky** [comment]
Whoa! That's pretty nice. Very readable and understandable in my opinion, too. particularly with the `head`, `tail` names.

**Comment by majjoha** [comment]
Thanks! I really like it too. I've been using a lot of languages that support pattern matching lately, where I'd often match on an empty list or extract the head from the tal with `head :: tail`. I think it is nice that Ruby supports a similar construct.

**Comment by tejasbubane** [comment]
This is cool :+1:

**Comment by maxehmookau** [comment]
Seriously cool. :open_mouth:

**Comment by kruszczynski** [comment]
That's cool!

**Comment by mike-burns** [comment]
This has the same effect as `#drop` on the tail when applied to the empty array, but if you assign the first value it might be worth mentioning: it will be `nil` if the array is empty.

**Comment by majjoha** [comment]
Good point, @mike-burns. I've updated the PR with your comment.

File: ruby/all-but-the-first-element-from-array.md [hunk 1]
```diff
@@ -21,6 +21,35 @@ rest_of_list = list
...
+head
+#=> 1
+tail
+#=> [2,3,4]
+
+list # is not modified
+#=> [1,2,3,4]
+```
+
+This has the same effect as `#drop` on the tail when applied to the empty array.
```
Comment (JoelQ) [inline ruby/all-but-the-first-element-from-array.md]: `drop` hasn't been mentioned yet. Thoughts on rewording it to just say the `tail` will not be `nil` when used with en empty array?

Comment (majjoha) [inline ruby/all-but-the-first-element-from-array.md]: @JoelQ You're right. I've updated the wording. Please, let me know if it needs any further changes.

File: ruby/all-but-the-first-element-from-array.md [hunk 2]
```diff
@@ -21,6 +21,34 @@ rest_of_list = list
...
+head
+#=> 1
+tail
+#=> [2,3,4]
+
+list # is not modified
+#=> [1,2,3,4]
+```
+
+When using parallel assignment on an empty array, the tail will not be `nil`.
```
Comment (JoelQ) [inline ruby/all-but-the-first-element-from-array.md]: Thoughts on moving this to the bottom, combining it with the similar section from `drop`, and creating a new section (perhaps named "Empty Arrays") that shows this behavior?

Comment (majjoha) [inline ruby/all-but-the-first-element-from-array.md]: It's completely fine with me. I've moved the example down, and I now mention `#drop` again in the section about empty arrays.

**Comment by derekprior** [comment]
This seems like a new TIL to me. We learned another way to do this.

**Comment by majjoha** [comment]
It could be separated into its own TIL, however I think it makes sense to include it in this document. What do you think, @JoelQ?

**Comment by JoelQ** [comment]
Although it certainly makes sense as part of this document, parallel assignment could definitely stand on it's own. These tips don't need to cover all the different options.

My preference would be a new tip, perhaps linking back to this one.

**Comment by majjoha** [comment]
I've now moved the tip into its own document and updated the description a bit too. I hope it looks good to you.

**Comment by JoelQ** [comment]
I like it! Thanks for contributing @majjoha. I'd like to credit you when we tweet this, what is your twitter handle?

**Comment by majjoha** [comment]
Thank you! It is @majjoha, so exactly the same as I use here on GitHub.

**Comment by JoelQ** [comment]
Merged in e3a81f1f6035f30a012866db872d114bd2c4fa7a

**Comment by manojkumarshukla** [comment]
@MANOKUMARSHUKLA

On Mon, Jan 26, 2015 at 11:07 PM, Joël Quenneville <notifications@github.com
> wrote:

> I like it! Thanks for contributing @majjoha <https://github.com/majjoha>.
> I'd like to credit you when we tweet this, what is your twitter handle?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/thoughtbot/til/pull/19#issuecomment-71501660>.
>



-- 

Manoj Kumar Shukla
   www.srvsewa.org
    +91-9407706543
    +91-9893916543
        07297-277001

**Pull request closed without merging**